pkg/epp/datastore/datastore.go

```diff
@@ -116,6 +117,14 @@ func (ds *datastore) PoolSet(ctx context.Context, reader client.Reader, pool *v1
 
 	oldPool := ds.pool
 	ds.pool = pool
+	if oldPool == nil || pool.Spec.TargetPorts[0] != oldPool.Spec.TargetPorts[0] {
+		if datalayer.Enabled(logger) {
+			if source, found := datalayer.GetNamedSource(dlmetrics.DataSourceName); found {
```

Comment:
strongly recommend to update `GerNamedSource` function to use generics.
otherwise, on every call we would need to do type conversion.

see example here:
https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/pkg/epp/plugins/plugin_state.go#L133